### PR TITLE
Infiltration game mode

### PR DESCRIPTION
**Pull request by Absolucy**
:cl: Absolucy
rscadd: Added the Infiltration game mode!
rscadd: In Infiltration, there's a team of Syndicate infiltrators who will attempt to infiltrate Space Station 13.
rscadd: The infiltrators can have various objectives, ranging from simple traitor objectives, to kidnapping someone important, hijacking the AI unit, or stealing power!
/:cl:

## Infiltrator-specific objectives:

Infiltrators will always have 1 or 2 of these.

 - **Hijack an AI**: A special device spawns with them, they gotta slap an active AI core to install it. After a while, the AI gains malf powers and is subverted to the syndicate's will. The process is not instant, and is *extremely obvious* to the AI. Prepare for all hell to break loose as a horde of shitcurity comes to detach the hijackinator. Maybe you should just subvert it normally first, bumrushing the core will probably blow your cover and fuck your entire team over.
 - **Kidnap someone**: You have to have someone on the infiltrator ship or base at the end of the round. Heads of staff are most likely to be the target, followed by members of security, with normal crew members having the lowest priority when it comes to choosing a target.
 	- If a kidnap target suicides or ghosts, the objective will count as complete. This both prevents the target from stopping their greentext out of spite, and also lets them freely suicide/ghost after being kidnapped while the infiltrators greentext.
 - **Steal power**: Plant a powersink on the station and have it draw a certain amount of power. It'll automatically alert over the syndie channel and turn off whenever the objective has been reached.

Base/shuttle: 
image

## Notes

 - Infiltrators do not have access to "murderboney" uplink items, such as e-swords, large bombs, mantis blades, gloves of the north star, CQC, etc.

### CODE REVIEW

**Comment by maxion12345** [comment]
Should make it so they'll green text if the target ghosts/suicides only on the shuttle, otherwise they get a new target. That way they can avoid searching for someone who might have cyroed, or some assistant that killed himself roundstart.

**Comment by Hopekz** [comment]
> Should make it so they'll green text if the target ghosts/suicides only on the shuttle, otherwise they get a new target. That way they can avoid searching for someone who might have cyroed, or some assistant that killed himself roundstart.

Tbh it might be better to just leave this in to get these types of people banned

File: code/datums/hud.dm [hunk 1]
```diff
@@ -38,6 +38,7 @@ GLOBAL_LIST_INIT(huds, list(
 	ANTAG_HUD_HERETIC = new/datum/atom_hud/antag/hidden(),
 	ANTAG_HUD_MINDSLAVE = new/datum/atom_hud/antag/hidden(),
 	ANTAG_HUD_ZOMBIE = new/datum/atom_hud/antag(),
+	ANTAG_HUD_INFILTRATOR = new/datum/atom_hud/antag(), // Yogs
```
Comment (JamieD1) [inline code/datums/hud.dm]:
```suggestion
	ANTAG_HUD_INFILTRATOR = new/datum/atom_hud/antag() // Yogs
```

File: code/__DEFINES/{yogs_defines}/antagonists.dm [hunk 2]
```diff
@@ -1,8 +1,10 @@
-#define ANTAG_DATUM_VAMPIRE		/datum/antagonist/vampire
-#define ANTAG_DATUM_THRALL		/datum/antagonist/thrall
-#define ANTAG_DATUM_SLING		/datum/antagonist/shadowling
-#define ANTAG_DATUM_DARKSPAWN	/datum/antagonist/darkspawn
-#define ANTAG_DATUM_VEIL		/datum/antagonist/veil
+#define ANTAG_DATUM_VAMPIRE			/datum/antagonist/vampire
+#define ANTAG_DATUM_THRALL			/datum/antagonist/thrall
+#define ANTAG_DATUM_SLING			/datum/antagonist/shadowling
+#define ANTAG_DATUM_DARKSPAWN		/datum/antagonist/darkspawn
```
Comment (JamieD1) [inline code/__DEFINES/{yogs_defines}/antagonists.dm:4]:
```suggestion
#define ANTAG_DATUM_DARKSPAWN			/datum/antagonist/darkspawn
```

File: yogstation/code/game/gamemodes/infiltration/infiltration.dm [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			var/datum/mind/new_sit = pick_n_take(antag_candidates)
+			pre_sit += new_sit
+			new_sit.assigned_role = "Syndicate Infiltrator"
+			new_sit.special_role = "Syndicate Infiltrator"
+			log_game("[key_name(new_sit)] has been selected as a syndicate infiltrator")
+		return TRUE
+	else
+		setup_error = "Not enough infiltrator candidates"
+		message_admins("Not enough infiltrator candidates! Was making [n_agents], but we need [required_enemies]!")
+		return FALSE
```
Comment (JamieD1) [inline yogstation/code/game/gamemodes/infiltration/infiltration.dm]:
```suggestion
	setup_error = "Not enough infiltrator candidates"
	message_admins("Not enough infiltrator candidates! Was making [n_agents], but we need [required_enemies]!")
	return FALSE
```

**Review by JamieD1 — DISMISSED** [review]
Minor things,

Shouldnt stop a test merge.

Will do a proper look later.

File: code/__DEFINES/role_preferences.dm [hunk 4]
```diff
@@ -79,6 +80,7 @@ GLOBAL_LIST_INIT(special_roles, list(
 	ROLE_GANG = /datum/game_mode/gang, // yogs
 	ROLE_HERETIC = /datum/game_mode/heretics,
 	ROLE_HIVE = /datum/game_mode/hivemind,
+	ROLE_INFILTRATOR, = /datum/game_mode/infiltrator, // yogs
```
Comment (JamieD1) [inline code/__DEFINES/role_preferences.dm]:
```suggestion
	ROLE_INFILTRATOR = /datum/game_mode/infiltrator, // yogs
```

File: code/__DEFINES/role_preferences.dm [hunk 5]
```diff
@@ -79,6 +80,7 @@ GLOBAL_LIST_INIT(special_roles, list(
 	ROLE_GANG = /datum/game_mode/gang, // yogs
 	ROLE_HERETIC = /datum/game_mode/heretics,
 	ROLE_HIVE = /datum/game_mode/hivemind,
+	ROLE_INFILTRATOR = /datum/game_mode/infiltrator, // yogs
```
Comment (JamieD1) [inline code/__DEFINES/role_preferences.dm]:
```suggestion
	ROLE_INFILTRATOR = /datum/game_mode/infiltration, // yogs
```

File: _maps/templates/infiltrator_base.dmm [hunk 6]
```diff
@@ -0,0 +1,3104 @@
...
+/turf/closed/indestructible/riveted,
+/area/yogs/infiltrator_base/jail)
+"bq" = (
+/turf/closed/indestructible/fakedoor{
+	name = "Primary Hallway"
+	},
+/area/yogs/infiltrator_base/jail)
+"br" = (
+/obj/machinery/vending/cigarette{
+	products = list(/obj/item/storage/box/fancy/cigarettes/cigpack_syndicate = 7, /obj/item/storage/fancy/cigarettes/cigpack_robust = 2, /obj/item/storage/fancy/cigarettes/cigpack_carp = 3, /obj/item/storage/fancy/cigarettes/cigpack_midori = 1, /obj/item/storage/box/matches = 10, /obj/item/lighter/greyscale = 4, /obj/item/storage/fancy/rollingpapers = 5)
```
Comment (JamieD1) [inline _maps/templates/infiltrator_base.dmm]:
```suggestion
	products = list(/obj/item/storage/box/fancy/cigarettes/cigpack_syndicate = 7, /obj/item/storage/fancy/cigarettes/cigpack_carp = 3, /obj/item/storage/fancy/cigarettes/cigpack_midori = 1, /obj/item/storage/box/matches = 10, /obj/item/lighter/greyscale = 4, /obj/item/storage/fancy/rollingpapers = 5)
```

**Comment by Absolucy** [comment]
I made it so killing the AI will abort a hijack in process, or un-hijack it if it is hijacked. So yes, you can kill then revive a hijacked AI to return it back to normal.

File: code/modules/clothing/chameleon.dm [hunk 7]
```diff
@@ -212,7 +212,7 @@
 		thing.update_slot_icon()
 	UpdateButtonIcon()
 
-/datum/action/item_action/chameleon/change/proc/update_item(obj/item/picked_item)
+/datum/action/item_action/chameleon/change/proc/update_item(obj/item/picked_item, obj/item/target = target) //yogs -- add support for cham hardsuits
```
Comment (adamsong) [inline code/modules/clothing/chameleon.dm]:
```suggestion
/datum/action/item_action/chameleon/change/proc/update_item(obj/item/picked_item) //yogs -- add support for cham hardsuits
```
The second argument causes a runtime

**Review by adamsong — DISMISSED** [review]

File: code/modules/clothing/chameleon.dm (hunk 7 above)
Comment (Absolucy) [reply code/modules/clothing/chameleon.dm]: ... what's the runtime? also there's a reason that argument was added.

Comment (Absolucy) [reply code/modules/clothing/chameleon.dm]: fixed. it needed to be `target = src.target`

**Comment by Mqiib** [comment]
I wouldn't say CQC is a very "murderboney" syndie item considering it's actually BEST at nonlethal takedowns since it does solely stamina and oxyloss damage (to a limit) as well as being able to silence and knock people out for a decently long time. Also very stealthy since it's a martial art.

**Comment by Absolucy** [comment]
I have several ideas for the shuttle, please react with which one y'all would prefer:

👍	Keep it as-is, with turrets and stuff.
👎	Replace turrets with toggleable energy shield surround the shuttle, which makes breaking into the shuttle difficult.
🚀	Replace turrets with a cloak for the shuttle, making it near-invisible when on.
👀	Replace turrets with the ability for the shuttle to go into transit and stay there, and then be called to any location in space by the infiltrators.

**Review by patpol4 — APPROVED** [review]

**Comment by Absolucy** [comment]
Alright, so infiltrators have their own custom implant now. They get an integrated pinpointer, similar to internal affairs agents, except you can click on the pinpointer to open your uplink, change the pinpointer target, or send the ship back to the base or to  the station.

No cloak - too difficult to properly implement, plus it'd be easily powergamed.

**Comment by Absolucy** [comment]
I give up. This mode will never work, everyone's too damn stupid to make it work.

**Comment by Cartlord** [comment]
I'm really not lking this in practice, it just feels like a greenshift until you suddenly lose the round because people you didn't know existed did some arbitrary shit. I've seen many people in OOC just call it "nukies lite" or somesuch, though I don't necessarily agree (for one thing, Nukies can be noticed).
literally while i was typing this, the pr was closed - wow

**Comment by SomeguyManperson** [comment]
> I give up. This mode will never work, everyone's too damn stupid to make it work.

people just haven't played it enough

**Comment by Hopekz** [comment]
> I give up. This mode will never work, everyone's too damn stupid to make it work.

Welcome to yogs.

Imo game design should be about catering to the type of players you want to see in the game vs the current playerbase.

**Comment by Absolucy** [comment]
Reopening because I've seen many people actually want this, which is kinda nice.

**Comment by Absolucy** [comment]
Merge conflict fixed

plus added a new codersprite for the infiltrator pinpointer, meant to make it more obvious that it's a whole thing you can interact with
image

please someone make a better sprite

**Review by patpol4 — DISMISSED** [review]
You appear to have cut off a wall from the ctf arena, please bring it back
image

**Comment by Absolucy** [comment]
Whoops, copy-paste strikes again. Will fix when I get home.

**Comment by patpol4** [comment]
https://github.com/yogstation13/Yogstation/issues/12725

**Comment by nmajask** [comment]
Might be nice to have them as a ghost visible antag to make it easier for observers to follow along, as similar antags that are easily discovered by ghosts (nukies, clock cultists, abductors) are.

**Comment by adamsong** [comment]
Please fix the merge conflicts

**Review by SomeguyManperson — APPROVED** [review]

**Comment by Addust** [comment]
infiltrator ship is a cooler pirate cutter

**Pull request merged**